internal/fs/fs.go

```diff
@@ -1972,6 +1972,45 @@ func (fs *fileSystem) renameFile(
 	return nil
 }
 
+func (fs *fileSystem) handlePendingInodes(inodes *[]inode.DirInode) func() {
+	return func() {
+		for _, in := range *inodes {
+			fs.unlockAndDecrementLookupCount(in, 1)
+		}
+		*inodes = []inode.DirInode{}
+	}
+}
+
+func (fs *fileSystem) getBucketDirInode(ctx context.Context, parent inode.DirInode, name string, pendingInodes *[]inode.DirInode) (inode.BucketOwnedDirInode, error) {
+	dir, err := fs.lookUpOrCreateChildDirInode(ctx, parent, name)
+	if err != nil {
+		return nil, fmt.Errorf("lookup directory: %w", err)
+	}
+	*pendingInodes = append(*pendingInodes, dir)
+	return dir, nil
+}
+
+func (fs *fileSystem) checkAndHandleLocalFiles(oldDir inode.BucketOwnedDirInode, oldName string) error {
+	fs.mu.Lock()
+	entries := oldDir.LocalFileEntries(fs.localFileInodes)
+	fs.mu.Unlock()
+	if len(entries) != 0 {
+		return fmt.Errorf("can't rename directory %s with open files: %w", oldName, syscall.ENOTSUP)
+	}
+	return nil
+}
+
+func (fs *fileSystem) checkNewDirNonEmpty(newDir inode.BucketOwnedDirInode, newName string) error {
```

Comment:
can you rename this to checkDirNotEmpty. Putting new in the name and the parameters is not required.

Reply:
Done